src/StoreApi/Formatters.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\StoreApi;
+
+use \Exception;
+use Automattic\WooCommerce\Blocks\StoreApi\Formatters\AbstractFormatter;
+use Automattic\WooCommerce\Blocks\StoreApi\Formatters\DefaultFormatter;
+
+/**
+ * Formatters class.
+ *
+ * Allows formatter classes to be registered. Formatters are exposed to extensions via the ExtendRestApi class.
+ */
+class Formatters {
+	/**
+	 * Holds an array of formatter classes.
+	 *
+	 * @var array
+	 */
+	private $formatters = [];
+
+	/**
+	 * Get a new instance of a formatter class.
+	 *
+	 * @throws Exception An Exception is thrown if a non-existing formatter is used and the user is admin.
+	 *
+	 * @param string $name Name of the formatter.
+	 * @return AbstractFormatter Formatter class instance.
+	 */
+	public function __get( $name ) {
+		if ( ! isset( $this->formatters[ $name ] ) ) {
+			if ( defined( 'WP_DEBUG' ) && WP_DEBUG && current_user_can( 'manage_woocommerce' ) ) {
+				throw new Exception( $name . ' formatter does not exist' );
+			}
+			return new DefaultFormatter();
+		}
+		return new $this->formatters[ $name ]();
```

Comment:
What's the intention of lazy instantiation here? Is there any benefit over just registering the formatter instance (and thus allowing to fail quicker if the instance registered is not of `FormatterInterface`)?

Reply:
I didn't want code that set options to interfere with other usage.

Reply:
Ohhh, I see, and this applies to my other comment which would be irrelevant because options are set per instance. I think I'd prefer the options to be configured on the fly with an argument on `format` as it seems like it'd be a more straightforward API (and there'd only ever be one instance of the formatter in a request).